src/clj/puppetlabs/services/jruby/jruby_puppet_schemas.clj

```diff
@@ -172,3 +181,12 @@
                         retry-poison-pill?
                         jruby-puppet-instance?)))
 
+(def JRubyMain
+  (schema/pred jruby-main-instance?))
+
+(def JRubyMainStatus
+  (schema/pred jruby-main-status-instance?))
+
+(def JRubyMainStatusOrNil
+  (schema/either JRubyMainStatus
+    (schema/pred nil? 'nil?)))
```

Comment:
`maybe`...  I'm sure you're right, I just started writing schemas for functions so I'll check and fix it up if so.